src/util.ts

```diff
@@ -481,3 +481,27 @@ export function guessPackageNameFromFile(filePath): Promise<string> {
 		});
 	});
 }
+
+// This will check whether it's regular package or internal package
+// Regular package will always allowed
+// Internal package only allowed if the package doing the import is within the tree rooted at the parent of "internal" directory
+// see: https://golang.org/doc/go1.4#internalpackages
+// see: https://golang.org/s/go14internal
+export function isAllowToImportPackage(toDirPath: string, currentWorkspace: string, pkgPath: string, goVersion: SemVersion) {
```

Comment:
Shouldn't this logic be inside `getImportablePackages()` method?

Reply:
I wish... but currently the completion using `this.pkgsList` cache in `GoCompletionItemProvider`.

Import internal package behavior works differently to each file (resident in a package) that try to import. So the set of packages would not be the same to each importing package. See https://github.com/Microsoft/vscode-go/pull/1257/files#diff-52de61e365f975d5f753902051296b23R289

Reply:
Somehow not waiting for the `getImportablePackage` give illusion the completion was fast.. it just the completion would not give the full set of the packages for the first time. Somehow I like it because it fast. So if we stick with this, then the filtering need to be reside on the `GoCompletionItemProvider`.

Reply:
> Import internal package behavior works differently to each file (resident in a package) that try to import.

Its different for each package, but not for files in the same package correct?

The logic we have in place in `getImportablePackages` to return the relative paths for vendor pkgs also works differently to each package. Which is why it takes the `filePath` as a parameter.

If filtering for internals is inside the `GoCompletionProvider`, then the filtering for vendors should also be inside `GoCompletionProvider`

But I intentionally chose not to do so for vendors to keep things simple.

The cost of keeping the filtering out of `GoCompletionProvider` for users would be that the first time they edit a file inside a package, they might get suggestions from `vendor` or `internal` folders that they shouldn't get.

But the cache will catch up, and any further edits in the package will get self corrected.